src/PIL/DdsImagePlugin.py

```diff
@@ -148,15 +148,16 @@ def _open(self):
             self.tile = [("raw", (0, 0) + self.size, 0, (self.mode, 0, 1))]
         elif pfflags & DDPF_RGB:
             # Texture contains uncompressed RGB data
-            masks = {mask: ["R", "G", "B", "A"][i] for i, mask in enumerate(masks)}
-            rawmode = ""
             if pfflags & DDPF_ALPHAPIXELS:
-                rawmode += masks[0xFF000000]
+                mask_count = 4
             else:
                 self._mode = "RGB"
-            rawmode += masks[0xFF0000] + masks[0xFF00] + masks[0xFF]
+                mask_count = 3
 
-            self.tile = [("raw", (0, 0) + self.size, 0, (rawmode[::-1], 0, 1))]
+            masks = struct.unpack(
+                "<" + str(mask_count) + "I", header.read(mask_count * 4)
```

Comment:
More performant and more readable (IMO):
```suggestion
                f"<{mask_count}I", header.read(mask_count * 4)
```